# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/trunk/webrtc/voice_engine/channel.cc
# Commit: b94935661622
# Full Hash: b949356616223660996acf39e1e353e7bbd2a60d
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2018-11-03 09:44:53
# Regressor Bug: 1376873
# File Overlap Count: 4
# Description:
#   Bug 1376873 - Use Call interface in AudioConduit; r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D7441
# ==============================================================================

diff -r 6eca45c83131 -r b94935661622 media/webrtc/trunk/webrtc/voice_engine/channel.cc
--- a/media/webrtc/trunk/webrtc/voice_engine/channel.cc	Mon Mar 26 16:19:23 2018 -0400
+++ b/media/webrtc/trunk/webrtc/voice_engine/channel.cc	Tue Aug 21 13:39:53 2018 -0400
@@ -1475,6 +1475,18 @@
   return 0;
 }
 
+int Channel::SetReceiveCsrcAudioLevelIndicationStatus(bool enable,
+                                                      unsigned char id) {
+  rtp_header_parser_->DeregisterRtpHeaderExtension(kRtpExtensionCsrcAudioLevel);
+  if (enable &&
+      !rtp_header_parser_->RegisterRtpHeaderExtension(kRtpExtensionCsrcAudioLevel,
+                                                      id)) {
+    return -1;
+  }
+  return 0;
+}
+
+
 void Channel::EnableSendTransportSequenceNumber(int id) {
   int ret =
       SetSendRtpHeaderExtension(true, kRtpExtensionTransportSequenceNumber, id);
@@ -1496,6 +1508,10 @@
   TransportFeedbackObserver* transport_feedback_observer =
       transport->transport_feedback_observer();
   PacketRouter* packet_router = transport->packet_router();
+  //This allows us to re-create streams but keep the same channel
+  if (packet_router_ == packet_router) {
+    return;
+  }
 
   RTC_DCHECK(rtp_packet_sender);
   RTC_DCHECK(transport_feedback_observer);
@@ -1515,6 +1531,11 @@
 void Channel::RegisterReceiverCongestionControlObjects(
     PacketRouter* packet_router) {
   RTC_DCHECK(packet_router);
+  //This allows us to re-create streams but keep the same channel
+  if (packet_router_ == packet_router) {
+    return;
+  }
+
   RTC_DCHECK(!packet_router_);
   constexpr bool remb_candidate = false;
   packet_router->AddReceiveRtpModule(_rtpRtcpModule.get(), remb_candidate);